### PR TITLE
Exporting sie file from makeradmin

### PR DESCRIPTION
**Pull request by BerglundDaniel**
The goal of this PR is to implement a route that can be used for the export button in #396

It is dependent on #394

Leaving some TODOs in to be fixed next week since we need the bookeeping to be done

## Summary by CodeRabbit

- **New Features**
  - Introduced new accounting entry types for clearer transaction categorization.
  - Added time period functionality for better date range management.
  - Implemented SIE file generation for standardized accounting data export.
  - Expanded accounting capabilities with new transaction handling and export features.
  - Added new routes for bookkeeping data export and file downloads.

- **Enhancements**
  - Improved Stripe payment integration with better error handling and payment retrieval.

- **Bug Fixes**
  - Adjusted accounting data model to ensure correct fraction value handling.

- **Documentation**
  - Updated codebase to include more descriptive comments and documentation for accounting processes.

- **Tests**
  - Added comprehensive test cases for accounting functionality and SIE file generation.

### CODE REVIEW

File: api/src/shop/accounting/accounting.py [hunk 1]
```diff
@@ -20,20 +21,20 @@
...
-    type: str  # TODO fix
+    account: str | None
     cost_center: str | None
+    type: AccountingEntryType
 
 
 @dataclass()
 class AccountCostCenter:
-    acccount: str
+    acccount: str | None
```
Comment (emanuelen5) [inline api/src/shop/accounting/accounting.py:34]: Misspelling of "account"

Comment (BerglundDaniel) [reply api/src/shop/accounting/accounting.py]: Thanks, fixed

**Comment by BerglundDaniel** [comment]
@coderabbitai review

File: api/src/shop/accounting/test/accounting_test.py [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+        payment = diff[0][1]
+        assert payment.amount == correct_payment.amount
+        assert payment.amount != wrong_transaction.amount
+        assert transaction.amount == wrong_transaction.amount
+        assert payment.transaction_id == correct_payment.transaction_id
+        assert payment.transaction_id == wrong_transaction.id
+        assert transaction.id == wrong_transaction.id
+
+
+# TODO fix all the | None to be Optional[...] instead
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/shop/accounting/test/accounting_test.py:143]: Why should this be done? How is `Optional` better?

**Pull request merged**